### PR TITLE
config(engine): Refine meta storeConfig(Part1)

### LINKED ISSUE #6173: conifg(engine): refine storeConfig to support new isolation schema
### Is your feature request related to a problem?

No

### Describe the feature you'd like

Refine storeConfig to support new isolation schema

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by maxshuang**
### What problem does this PR solve?

Issue Number: close #6173 

### What is changed and how it works?
1. Add 'Schema' param to 'StoreConfig'
2. Unify 'StoreConfigParams' to 'StoreConfig'
3. Extract 'dbutil' pkg to make sql implement for kv client easier 

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

File: engine/pkg/dbutil/util.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	"github.com/pingcap/log"
+	cerrors "github.com/pingcap/tiflow/pkg/errors"
+	"go.uber.org/zap"
+)
+
+// NewSQLDB return sql.DB for specified driver and dsn
+func NewSQLDB(driver string, dsn string, dbConf *DBConfig) (*sql.DB, error) {
+	db, err := sql.Open(driver, dsn)
+	if err != nil {
+		log.L().Error("open dsn fail", zap.String("dsn", dsn), zap.Any("config", dbConf), zap.Error(err))
```
Comment (CharlesCheung96) [inline engine/pkg/dbutil/util.go:29]:
```suggestion
		log.Error("open dsn fail", zap.String("dsn", dsn), zap.Any("config", dbConf), zap.Error(err))
```

File: engine/pkg/orm/client.go [hunk 2]
```diff
@@ -142,14 +142,14 @@ type ResourceClient interface {
 }
 
 // NewClient return the client to operate framework metastore
-func NewClient(mc metaclient.StoreConfigParams, conf DBConfig) (Client, error) {
+func NewClient(mc metaclient.StoreConfig, conf dbutil.DBConfig) (Client, error) {
```
[the remark this answers is not in the document]
Comment (maxshuang) [reply engine/pkg/orm/client.go:145]: I have done it in anther pr.
So we can ignore it temporary.

File: engine/pkg/meta/metaclient/config.go [hunk 3]
```diff
@@ -14,46 +14,86 @@
...
+	}
+	if storeConf.Auth != nil {
+		dsnCfg.User = storeConf.Auth.User
+		dsnCfg.Passwd = storeConf.Auth.Passwd
 	}
+	dsnCfg.Net = "tcp"
+	dsnCfg.Addr = storeConf.Endpoints[0]
+	dsnCfg.DBName = storeConf.Schema
+	dsnCfg.InterpolateParams = true
+	// dsnCfg.MultiStatements = true
```
[the remark this answers is not in the document]
Comment (maxshuang) [reply engine/pkg/meta/metaclient/config.go:90]: Ok.

File: engine/pkg/dbutil/config.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+package dbutil
+
+import "time"
+
+const (
+	defaultConnMaxIdleTime = 1 * time.Minute
+	defaultConnMaxLifeTime = 12 * time.Hour
+	defaultMaxIdleConns    = 1
+	defaultMaxOpenConns    = 1
```
[the remark this answers is not in the document]
Comment (maxshuang) [reply engine/pkg/dbutil/config.go:22]: At first, I am planning to create one connection for a client.
But now, I think split a common db connection layer may be better. 
I will make it larger.

File: engine/pkg/meta/metaclient/config.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (maxshuang) [reply engine/pkg/meta/metaclient/config.go]: Done

**Review by amyangfei — APPROVED** [review]

File: engine/pkg/dbutil/util.go [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+	db.SetConnMaxLifetime(dbConf.ConnMaxLifeTime)
+	db.SetMaxIdleConns(dbConf.MaxIdleConns)
+	db.SetMaxOpenConns(dbConf.MaxOpenConns)
+	return db, nil
+}
+
+// GetSQLStrictMode return SQL strict mode for metastore
+func GetSQLStrictMode() string {
+	needEnable := []string{
+		"STRICT_TRANS_TABLES",
```
Comment (CharlesCheung96) [inline engine/pkg/dbutil/util.go:43]: Could you add comments here?

Comment (maxshuang) [reply engine/pkg/dbutil/util.go:44]: Done.

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**